assets/runtime/config/gitlabhq/gitlab.yml

```diff
@@ -493,6 +493,10 @@ production: &base
       default:
         path: {{GITLAB_REPOS_DIR}}/
         gitaly_address: unix:/home/git/gitlab/tmp/sockets/private/gitaly.socket # TCP connections are supported too (e.g. tcp://host:port)
+        failure_count_threshold: 10 # number of failures before stopping attempts
```

Comment:
The env varible cloud be `GITLAB_ REPOSITORIES_STORAGES_DEFAULT_FAILURE_COUNT_THRESHOLD`. The default value would be `10`.